resource-managers/yarn/src/main/scala/org/apache/spark/deploy/yarn/ApplicationMaster.scala

```diff
@@ -67,6 +68,14 @@ private[spark] class ApplicationMaster(args: ApplicationMasterArguments) extends
 
   private val securityMgr = new SecurityManager(sparkConf)
 
+  private[spark] val failureTracker = new FailureTracker(sparkConf, new SystemClock)
```

Comment:
No need to specify the clock here?

Reply:
As other metrics from YarnAllocator will be added this change is reverted.